test/slowmo.spec.ts

```diff
@@ -0,0 +1,237 @@
+/**
+ * Copyright (c) Microsoft Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import './base.fixture';
+
+import { attachFrame } from './utils';
+
+async function checkSlowMo(toImpl, page, task) {
+  let didSlowMo = false;
+  const orig = toImpl(page)._doSlowMo;
+  toImpl(page)._doSlowMo = async function (...args) {
+    if (didSlowMo)
+      throw new Error('already did slowmo');
+    await new Promise(x => setTimeout(x, 100));
+    didSlowMo = true;
+    return orig.call(this, ...args)
+  }
+  await task();
+  expect(!!didSlowMo).toBe(true);
+}
+
+async function checkPageSlowMo(toImpl, page, task) {
+  await page.setContent(`
+    <button>a</button>
+    <input type="checkbox" class="check">
+    <input type="checkbox" checked=true class="uncheck">
+    <input class="fill">
+    <select>
+      <option>foo</option>
+    </select>
+    <input type="file" class="file">
+  `)
+  await checkSlowMo(toImpl, page, task);
+}
+
+it('Page SlowMo $$eval', async ({page, toImpl}) => {
```

Comment:
You'll have to mark all of these as `skip(WIRE)`, because toImpl is not available there.